asv/benchmarks/dwt_benchmarks.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import pywt
+
+
+class DwtTimeSuiteBase(object):
+    """
+    Set-up for (I)DWT timing.
+    """
+    params = ([16, 100, 101, 256, 490, 491, 512],
+              ['haar', 'db4', 'sym20'],
+              pywt.Modes.modes)
+    param_names = ('n', 'wavelet', 'modes')
+
+    def setup(self, n, wavelet, mode):
+        self.data = np.ones(n, dtype='float')
+        self.wavelet = pywt.Wavelet(wavelet)
+        self.mode = pywt.Modes.from_object(mode)
+
+
+class DwtTimeSuite(DwtTimeSuiteBase):
+    def time_dwt(self, n, wavelet, mode):
+        pywt.dwt(self.data, wavelet, mode)
+
+    def time_dwt_ext(self, n, wavelet, mode):
+        pywt._extensions._dwt.dwt_single(self.data, self.wavelet, self.mode)
```

Comment:
This probably shouldn't be in a benchmark - only need/want to benchmark public functions/methods.